dask/dataframe/tests/test_groupby.py

```diff
@@ -2143,6 +2143,27 @@ def test_groupby_shift_lazy_input():
         )
 
 
+@pytest.mark.filterwarnings("ignore:`meta` is not specified")
+def test_groupby_shift_within_partition_sorting():
+    # Result is non-deterministic. We run the assertion a few times to keep
+    # the probability of false pass low.
+    for _ in range(10):
+        df = pd.DataFrame(
+            {
+                "a": range(60),
+                "b": [2, 4, 3, 1] * 15,
+                "c": [None, 10, 20, None, 30, 40] * 10,
+            }
+        )
+        df = df.set_index("a").sort_index()
+        ddf = dd.from_pandas(df, npartitions=6)
+        assert_eq(
+            df.groupby("b")["c"].shift(1),
+            ddf.groupby("b")["c"].shift(1),
```

Comment:
We drop the `compute` and `sort_index` calls from here, letting `assert_eq` handle that (by default `assert_eq` will sort by the index). Letting `assert_eq` handle the compute is important, as it also checks that the metadata on the dask collection (the dtypes/divisions/etc...) match the computed result. If you call compute beforehand then `assert_eq` only sees two pandas dataframes and can't perform these checks.